_posts/2025-05-23-5.12.0-released.md

```diff
@@ -0,0 +1,113 @@
+  Luanti 5.12.0 is here! Perfectly on-schedule, we swear! Anyway, this time Luanti is sporting new input handling and controls, new item and entity visuals, better map performance, and more.
+excerpt: >-
+  Luanti 5.12.0 is here! Perfectly on-schedule, we swear! Anyway, this time Luanti is sporting new input handling and controls, new item and entity visuals, better map performance, and more.
+image: /static/blog/5.12.0/cover.webp
+image_credit: 
+forum_topic: https://forum.luanti.org/viewtopic.php?t=31686
+tags:
+  - release
+---
+
+<h2 class="sr-only">Introduction</h2>
+
+Luanti 5.12.0 is here! Perfectly on schedule, we swear! Anyway, this time Luanti is sporting new input handling and controls, new item and entity visuals, better map performance, and more.
+
+You can download Luanti from [our website](https://www.luanti.org/downloads/), check out the code on [GitHub](https://github.com/luanti-org/luanti/), or read the [full changelog](https://docs.luanti.org/changelog/#5110--5120).
+
+### Table of Contents
+
+- [SDL2 Window and Input Handling](#sdl2-window-and-input-handling)
+- [Control Your Controls](#control-your-controls)
+- [More Animation, Less Blur, and New Reviews](#more-animation-less-blur-and-new-reviews)
+- [Camera Mode, Entity Visuals, and More](#camera-mode-entity-visuals-and-more)
+- [World and Object Performance](#world-and-object-performance)
+- [Other Highlights](#other-highlights)
+- [What's Next?](#whats-next)
+
+## SDL2 Window and Input Handling
+
+This section covers a plethora of outstanding issues, so don't skip this one— SDL is a bigger deal than you might think! Until this release, Luanti has relied on Irrlicht's custom window and input interfaces, which (much like a lot of old Irrlicht code) was _very_ outdated and buggy. [SDL](https://en.wikipedia.org/wiki/Simple_DirectMedia_Layer), on the other hand, has been consistently updated since 1998 and has been battle-tested by thousands of major applications over the years.
+
+Efforts to migrate to SDL2 have been ongoing for a few years now, and 5.12.0 is the first version to finally see those efforts come to fruition. The cross-platform wrappers SDL2 brings have enabled a bunch of cool stuff:
+* Touchscreen controls are now supported on any touchscreen device (not just mobile) ([irr#262](https://github.com/minetest/irrlicht/pull/262), [#14146](https://github.com/luanti-org/luanti/pull/14146), [#14075](https://github.com/luanti-org/luanti/pull/14075), others)
+* Keybinds now use [scancodes](https://en.wikipedia.org/wiki/Scancode), allowing non-QWERTY keyboards to work out of the box and fixing some platform-specific issues on macOS and others ([#14964](https://github.com/luanti-org/luanti/pull/14964))
+* Support for high-DPI screens ([#14703](https://github.com/luanti-org/luanti/pull/14703))
+* Desktop [IME](https://en.wikipedia.org/wiki/Input_method) support for text input (helpful for CJK and other lanuages) ([#14942](https://github.com/luanti-org/luanti/pull/14942))
```

Comment:
Nitpick since IME support for the X11 device was added in 2021 (minetest/irrlicht#23)

```suggestion
* Desktop [IME](https://en.wikipedia.org/wiki/Input_method) support for text input on all platforms (helpful for CJK and other lanuages) ([#14942](https://github.com/luanti-org/luanti/pull/14942))
```